test/RelayHubGasCalculations.test.ts

```diff
@@ -337,6 +345,88 @@ contract('RelayHub gas calculations', function ([_, relayOwner, relayWorker, rel
+      const gassesUsed: any[] = [];
+      [0, 32, 128, 16384/* , 32768, 65536 */].forEach(dataLength => {
+        it(`with arg: ${dynamicArg} length: ${dataLength}`, async function () {
+          // console.log('gasUsed: ', gassesUsed)
+          const id = (await snapshot()).result
+          const senderNonce = (await forwarderInstance.getNonce(senderAddress)).toString()
+          let approvalData = '0x'
+          let paymasterData = '0x'
+          let encodedFunction = recipient.contract.methods.dontEmitMessage('').encodeABI()
+          if (dynamicArg === RelayCallDynamicArg.APPROVAL_DATA) {
+            approvalData = '0x' + 'ff'.repeat(dataLength)
+          } else if (dynamicArg === RelayCallDynamicArg.ENCODED_FUNCTION) {
+            // encoded function arg is an ascii string, so each char is a byte. We append half the length then.
+            encodedFunction = recipient.contract.methods.dontEmitMessage('f'.repeat(dataLength)).encodeABI()
+            // console.log('encodedFunction', encodedFunction)
+          } else if (dynamicArg === RelayCallDynamicArg.PAYMASTER_DATA) {
+            paymasterData = '0x' + 'ff'.repeat(dataLength)
+          }
+          const relayRequest: RelayRequest = {
+            request: {
+              to: recipient.address,
+              data: encodedFunction,
+              from: senderAddress,
+              nonce: senderNonce,
+              value: '0',
+              gas: gasLimit.toString()
+            },
+            relayData: {
+              baseRelayFee: '0',
+              pctRelayFee: '0',
+              gasPrice: '1',
+              relayWorker,
+              forwarder,
+              paymaster: paymaster.address,
+              paymasterData,
+              clientId
+            }
+          }
+          const dataToSign = new TypedRequestData(
+            chainId,
+            forwarder,
+            relayRequest
+          )
+          const signature = await getEip712Signature(
+            web3,
+            dataToSign
+          )
+          const relayCall = relayHub.contract.methods.relayCall(10e6, relayRequest, signature, approvalData, externalGasLimit)
+          const receipt = await relayCall.send({
+            from: relayWorker,
+            gas: externalGasLimit,
+            gasPrice: gasPrice
+          })
+          gassesUsed.push(receipt.gasUsed - calculateCalldataCost(relayCall.encodeABI()))
+          // console.log('relayCall encodeABI len', relayCall.encodeABI().length / 2)
+          // console.log('gasUsed is', receipt.gasUsed)
+          // console.log('calculateCalldataCost is', calculateCalldataCost(relayCall.encodeABI()))
+          if (gassesUsed.length > 1) {
+            const diff = gassesUsed[gassesUsed.length - 1] - gassesUsed[0]
+            // console.log('diff is', diff)
+            // console.log('diff per byte is', diff / dataLength)
+            if (diff / dataLength > hubDataGasCostPerByte) {
```

Comment:
there is an `assert.closeTo` check. I think it will be better fit as if our value is waay to big it may also be a problem

Reply:
Fixed